### PR TITLE
Fix `ios run` command for devices

### LINKED ISSUE #402: For real devices, `ios run` reports success even when app deployment failed
```
05:37:06.3854340 05:37:06.3854220 Xamarin.Hosting: Connected to Premek’s iPhone (00008030-000A08D234DA802E) in 00:00:00.0023261
05:37:06.5799000 05:37:06.5798890 Xamarin.Hosting: Connecting to 'Premek’s iPhone', token is 0x7fba22478f30
05:37:06.6436670 05:37:06.6436510 Xamarin.Hosting: Connected to 'Premek’s iPhone'
05:37:06.6441510 05:37:06.6441450 Xamarin.Hosting: Mounting developer image on 'Premek’s iPhone'
05:37:06.6612980 05:37:06.6612840 warning HE0030: Could not mount developer tools on 'Premek’s iPhone': The current device configuration is unsupported. This iPhone 11 is running iOS 14.2 (18B92), which is not supported by Xcode 1.0.
05:37:06.6613690 05:37:06.6613660 warning HE0031: Failed to mount developer tools on 'Premek’s iPhone'.
05:37:06.7363330 05:37:06.7363090 error MT1031: Could not launch the app 'net.dot.HelloiOS' on the device 'Premek’s iPhone' because the device is locked. Please unlock the device and try again.
05:37:06.7416610 05:37:06.7416480   at Xamarin.Launcher.DevController+<>c__DisplayClass14_0.<LaunchBundleOnDevice>b__0 () [0x005e2] in <02eb00c5c2a845edb9a0fcd1d11f9934>:0
05:37:06.7417170 05:37:06.7417150   at Xamarin.Launcher.DevController.LaunchDeviceBundleAsync (System.String app_path, Xamarin.Hosting.DeviceLaunchConfig config) [0x00111] in <02eb00c5c2a845edb9a0fcd1d11f9934>:0
05:37:06.7417340 05:37:06.7417320   at Xamarin.Utils.NSRunLoopExtensions.RunUntilTaskCompletion[T] (Foundation.NSRunLoop this, System.Threading.Tasks.Task`1[TResult] task) [0x00070] in <ef9b0748bc0b46b3925ad7585316745a>:0
05:37:06.7417600 05:37:06.7417580   at Xamarin.Launcher.Driver.Main2 (System.String[] args) [0x00b4d] in <02eb00c5c2a845edb9a0fcd1d11f9934>:0
05:37:06.7417750 05:37:06.7417740   at Xamarin.Launcher.Driver.Main (System.String[] args) [0x0006d] in <02eb00c5c2a845edb9a0fcd1d11f9934>:0
05:37:06.7491870 05:37:06.7491780 Process mlaunch exited with 1
05:37:11.7643440 05:37:11.7643320 Killing process tree of 3162...
05:37:11.7846320 05:37:11.7846140 Pids to kill: 3162
05:37:11.7857900 05:37:11.7857790 A capture of the device log is: /Users/prvysoky/Documents/tmp/o/device-Premek’s iPhone-20201218_053704.log
05:37:11.7886430 05:37:11.7886310 App run ended with 0
```

### PR DESCRIPTION
**Pull request by premun**
- `UninstallApp()` wasn't triggering for devices
- mlaunch failures when running app didn't get detected

Resolves #402

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.CLI/Commands/iOS/iOSAppCommand.cs [hunk 1]
```diff
@@ -159,24 +160,49 @@ private async Task<ExitCode> RunApp(ILogger logger, TestTargetOs target, Logs lo
...
                 }
-            }
+                catch (Exception e)
+                {
+                    installException = e;
+                }
+
+                if (exitCode != ExitCode.SUCCESS || installException != null)
+                {
+                    var message = new StringBuilder().AppendLine("Application installation failed:");
```
Comment (jakubstilec) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/iOS/iOSAppCommand.cs:180]: Wouldn't it be worth to extract error message logic? It's there twice with almost the same code.

File: src/Microsoft.DotNet.XHarness.iOS/AppRunner.cs [hunk 2]
```diff
@@ -230,23 +230,22 @@ public class AppRunner : AppRunnerBase
...
+                }
             }
             finally
             {
                 deviceLogCapturer.StopCapture();
                 deviceSystemLog.Dispose();
             }
-
-            // Upload the system log
-            if (File.Exists(deviceSystemLog.FullPath))
```
Comment (jakubstilec) [inline src/Microsoft.DotNet.XHarness.iOS/AppRunner.cs:246]: Why was this removed? Isn't it still needed?

Comment (premun) [reply src/Microsoft.DotNet.XHarness.iOS/AppRunner.cs:246]: Yeah, it was useless

**Review by jakubstilec — APPROVED** [review]

File: src/Microsoft.DotNet.XHarness.CLI/Commands/iOS/iOSAppCommand.cs (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/iOS/iOSAppCommand.cs:180]: I tried but the code is different enough (we use different knowledge base method, we get the exception from a different place, in one case we don't care about the exception message...) that it would be just too much overhead to add some common method for this.
